src/allmydata/test/test_eliotutil.py

```diff
@@ -95,6 +98,12 @@ def test_returns_unfired_deferred(self):
         # We didn't start an action.  We're not finishing an action.
         return d.result
 
+    def test_logs_non_utf_8_byte(self):
+        """
+        If an Eliot message is emitted that contains a non-UTF-8 byte string then
+        the test nevertheless passes.
+        """
+        Message.log(hello=b"\xFF")
```

Comment:
Is an assertion to see what is in the logger not necessary?

Reply:
To add to that... Eliot by design should never throw an exception from logging APIs. I find these tests confusing in general.

Reply:
The reason this test failed before the implementation changes is that:

1. The non-UTF-8 bytes could not be serialized because the test infrastructure didn't set up the any-bytes encoder and
2. `SyncTestCase` / `AsyncTestCase` runs Eliot message validation at the end of each test and fails the test if it fails

This certainly was not clear from the way these tests were written so I rewrote them to have explicit assertions and to have the case under test represented by a different test method than is making those assertions.

I rewrote the other tests in this case as well since the same reasoning applies to all of them and because it would have been more work to figure out how to keep them working as-is with the other changes necessary to do the above.